### PR TITLE
Fixed returning when updating compiler list

### PR DESCRIPTION
**Pull request by ADRFranklin**
Fixed issue with #315

### CODE REVIEW

File: download/compiler_cache.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func GetCompilerList(cacheDir string) (compilers types.Compilers, err error) {
 		fmt.Fprintln(os.Stderr, "updating compiler list...") // nolint:gas
 		err = UpdateCompilerList(cacheDir)
 		if err != nil {
-			return
+			fmt.Fprintln(os.Stderr, "failed to update compiler list")
```
Comment (Southclaws) [inline download/compiler_cache.go]: This should also print the `err` after the message and then set `err` to `nil` to clean up.

File: go.sum [hunk 2]
```diff
@@ -4,19 +4,15 @@ github.com/Masterminds/semver v1.4.2 h1:WBLTQ37jOCzSLtXNdoo8bNM8876KhNqOKvrlGITg
 github.com/Masterminds/semver v1.4.2/go.mod h1:MB6lktGJrhw8PrUyiEoblNEGEQ+RzHPF078ddwwvV3Y=
```
Comment (Southclaws) [inline go.sum]: Try and keep go.mod/sum changes out of PRs unless the PR explicitly requires it.

Comment (ADRFranklin) [reply go.sum]: Wait, you told me before to include it in PR's -_-. I remember asking you specifically about it.

File: download/compiler_cache.go (hunk 1 above)
Comment (ADRFranklin) [reply download/compiler_cache.go]: You said err should be set to nil to clean up, but the variable `err` is reused again at the end of the file and replaced? So does this still need cleaning up?

File: go.sum (hunk 2 above)
Comment (Southclaws) [reply go.sum]: No, go.mod/go.sum specify what dependencies the codebase requires, the changes you've made are not affected by any dependencies so these files shouldn't be changed during your patch.

Your editor/tooling might attempt to be helpful by auto updating dependencies by running `go mod tidy` in the background but this is not really useful. If you notice go.mod/sum files while running `git add` then reset their state back so your branch is using the same dependencies as upstream.

**Pull request closed without merging**